src/Core/src/Platform/Android/WrapperView.cs

```diff
@@ -171,79 +156,95 @@ void ClipChild(Canvas canvas)
 				canvas.ClipPath(_currentPath);
 		}
 
-		void DrawShadow(Canvas canvas)
+		int GetShadowHash()
 		{
-			if (_shadowCanvas == null)
-				_shadowCanvas = new Canvas();
+			if (Shadow == null)
+				return -1;
 
-			if (_shadowPaint == null)
-				_shadowPaint = new Android.Graphics.Paint
-				{
-					AntiAlias = true,
-					Dither = true,
-					FilterBitmap = true
-				};
+			return Shadow.GetHashCode();
```

Comment:
I'm very skeptical about this cache.

First, the cache instance should be static as it doesn't matter if 2 different views are using the "same" shadow (which is a likely situation if you're using a resource key) the same bitmap could be applied.

Second, what counts as the "same" shadow is not only the virtual views' shadow but also the dimensions. 2 views with exactly the same shadow properties, i.e same shadow hashcode, but different dimensions can't use the same bitmap.

Third, `InvalidateShadow` removes the shadow from the hash but that's not really what you want. It is called on every shadow change which means that there will only ever be 1 shadow in the cache and it is the current one. I don't think that's how caches are used 😅. Maybe a more complex cache like LRU-cache would help.